### PR TITLE
Added versioning support for git archives

### LINKED ISSUE #42: Add versioning support for git archives
Given that we are using `hatch-vcs` (in both the project and the template), the following two files should be added in to support versioning from VCS for a user installing the project through a git archive -

- `.git_archival.txt` ([example](https://github.com/scikit-hep/vector/blob/main/.git_archival.txt) (can be copied entirely))
- `.gitattributes` ([example](https://github.com/scikit-hep/vector/blob/main/.gitattributes) (can be copied entirely))

### PR DESCRIPTION
**Pull request by santacodes**
Fixes #42 

Added `.gitattributes.txt` and `git_archival.txt` to both project and template.

### CODE REVIEW

**Review by arjxn-py — APPROVED** [review]

File: template/{% if backend == 'hatch' and vcs == true %}.git_archival.txt{% endif %}.jinja [hunk 1]
```diff
@@ -0,0 +1,4 @@
+node: $Format:%H$
```
Comment (Saransh-cpp) [inline template/{% if backend == 'hatch' and vcs == true %}.git_archival.txt{% endif %}.jinja:1]: Should be present on `vcs == true` and any backend.

File: template/{% if backend == 'hatch' and vcs == true %}.gitattributes.txt{% endif %}.jinja [hunk 2]
```diff
@@ -0,0 +1 @@
+.git_archival.txt export-subst
```
Comment (Saransh-cpp) [inline template/{% if backend == 'hatch' and vcs == true %}.gitattributes.txt{% endif %}.jinja:1]: Ditto

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @santacodes! A couple of comments below -

**Review by Saransh-cpp — APPROVED** [review]
Thanks!

**Pull request merged**